### DISCUSSION TITLE
Checking request body with value of null from "Rules" return 500

### DISCUSSION THREAD

**Discussion by alvinchristian7** [mockoon/mockoon#409] [Help] [answered]
Hi all

As my title suggested, it always return **500 Internal Server Error** and its not from my custom response, but from **Mockoon** itself
I've put **Content-Type: application/json**, tried all possibility and worked except checking a variable with **null** value, i set it to imitate my real servers

Is there a workaround to this? Thanks

Request:
![request SS](https://user-images.githubusercontent.com/17759409/105943113-2a3c8380-6093-11eb-9316-e38a936b6376.png)

Req body:
`{
  pagination: null
}`

Responses copied to clipboard, problem in the 3rd:
`{
  "source": "mockoon:1.11.0",
  "data": [
      {
          "type": "route",
          "item": {
              "uuid": "",
              "documentation": "CC ACCOUNT LISTS v2",
              "method": "post",
              "endpoint": "api/myaccount/v1/getAccountListCC",
              "responses": [
                  {
                      "uuid": "",
                      "statusCode": 200,
                      "latency": 0,
                      "filePath": "",
                      "sendFileAsBody": false,
                      "headers": [
                          {
                              "key": "Content-Type",
                              "value": "application/json"
                          }
                      ],
                      "body": "{\r\n  \"myaccountresponse\": {\r\n        \"timestamp\": \"03-10-2019 05:31:27\",\r\n        \"message\": null,\r\n        \"traceId\": \"0daeeb51ae922243\",\r\n        \"status\": \"00\",\r\n        \"responseStatus\": \"SUCCESS\",\r\n        \"pagination\": \"Y\",\r\n        \"conventionalAccountList\": null,\r\n        \"shariaAccountList\": null,\r\n        \"accountDetail\": null,\r\n        \"sofConven\": null,\r\n        \"sofShariah\": null,\r\n        \"sofAll\": {\r\n            \"tdaccountList\": {},\r\n            \"ccaccountList\": {\r\n                {{#repeat 100}}\r\n               \"{{@index}}\": {\r\n                  \"accountIndex\": null,\r\n                  \"accountName\": \"Giro Individual\",\r\n                  \"accountNumber\": \"********8500\",\r\n                  \"accountStatus\": \"Active\",\r\n                  \"availableBalance\": 718500,\r\n                  \"accountType\": \"CDA\",\r\n                  \"description\": \"\",\r\n                  \"productId\": \"522866\",\r\n                  \"accountBalance\": 7185,\r\n                  \"currency\": \"IDR\",\r\n                  \"customerName\": \"CN003475 EMI\",\r\n                  \"allowed\": true,\r\n                  \"sharia\": false,\r\n                  \"applicationType\": \"B\"\r\n               }\r\n               {{/ repeat}}\r\n            },\r\n            \"caaccountList\": {},\r\n            \"saaccountList\": {},\r\n            \"nonaccountList\": {},\r\n            \"loanaccountList\": {}\r\n        }\r\n    }\r\n}",
                      "rules": [],
                      "label": "Repeat 100, Y",
                      "disableTemplating": false,
                      "rulesOperator": "OR"
                  },
                  {
                      "uuid": "",
                      "body": "{\r\n  \"myaccountresponse\": {\r\n        \"timestamp\": \"03-10-2019 05:31:27\",\r\n        \"message\": null,\r\n        \"traceId\": \"0daeeb51ae922243\",\r\n        \"status\": \"00\",\r\n        \"responseStatus\": \"SUCCESS\",\r\n        \"pagination\": \"N\",\r\n        \"conventionalAccountList\": null,\r\n        \"shariaAccountList\": null,\r\n        \"accountDetail\": null,\r\n        \"sofConven\": null,\r\n        \"sofShariah\": null,\r\n        \"sofAll\": {\r\n            \"tdaccountList\": {},\r\n            \"ccaccountList\": {\r\n               \"0\": {\r\n                  \"accountIndex\": null,\r\n                  \"accountName\": \"Giro Individual\",\r\n                  \"accountNumber\": \"********8500\",\r\n                  \"availableBalance\": 0,\r\n                  \"accountType\": \"CDA\",\r\n                  \"description\": \"\",\r\n                  \"productId\": \"522866\",\r\n                  \"accountBalance\": 0,\r\n                  \"currency\": \"IDR\",\r\n                  \"customerName\": \"CN003475 EMI\",\r\n                  \"allowed\": true,\r\n                  \"sharia\": false,\r\n                  \"applicationType\": \"B\"\r\n               },\r\n               \"1\":          {\r\n                  \"accountIndex\": null,\r\n                  \"accountName\": \"Giro Ind Corp\",\r\n                  \"accountNumber\": \"********5140\",\r\n                  \"availableBalance\": 403772,\r\n                  \"accountType\": \"CDA\",\r\n                  \"description\": \"\",\r\n                  \"productId\": \"467908\",\r\n                  \"accountBalance\": 4037.72,\r\n                  \"currency\": \"USD\",\r\n                  \"customerName\": \"CN003475 EMI\",\r\n                  \"allowed\": false,\r\n                  \"sharia\": false,\r\n                  \"applicationType\": null\r\n               }\r\n            },\r\n            \"caaccountList\": {},\r\n            \"saaccountList\": {},\r\n            \"nonaccountList\": {},\r\n            \"loanaccountList\": {}\r\n        }\r\n    }\r\n}",
                      "latency": 0,
                      "statusCode": 200,
                      "label": "2 accounts",
                      "headers": [
                          {
                              "key": "",
                              "value": ""
                          }
                      ],
                      "filePath": "",
                      "sendFileAsBody": false,
                      "rules": [],
                      "disableTemplating": false,
                      "rulesOperator": "OR"
                  },
                  {
                      "uuid": "",
                      "body": "{\r\n  \"myaccountresponse\": {\r\n        \"timestamp\": \"03-10-2019 05:31:27\",\r\n        \"message\": null,\r\n        \"traceId\": \"0daeeb51ae922243\",\r\n        \"status\": \"00\",\r\n        \"responseStatus\": \"SUCCESS\",\r\n        \"pagination\": \"N\",\r\n        \"conventionalAccountList\": null,\r\n        \"shariaAccountList\": null,\r\n        \"accountDetail\": null,\r\n        \"sofConven\": null,\r\n        \"sofShariah\": null,\r\n        \"sofAll\": {\r\n            \"tdaccountList\": {},\r\n            \"ccaccountList\": {\r\n               \"0\": {\r\n                  \"accountIndex\": 0,\r\n                  \"accountName\": \"Giro INDEX 0\",\r\n                  \"accountNumber\": \"4599{{ faker 'finance.account' 12 }}\",\r\n                  \"accountStatus\": \"Active\",\r\n                  \"availableBalance\": 0,\r\n                  \"accountType\": \"CDA\",\r\n                  \"description\": \"\",\r\n                  \"productId\": \"522866\",\r\n                  \"accountBalance\": 0,\r\n                  \"currency\": \"IDR\",\r\n                  \"customerName\": \"CN003475 EMI\",\r\n                  \"allowed\": true,\r\n                  \"sharia\": false,\r\n                  \"applicationType\": \"B\"\r\n               },\r\n               \"1\":          {\r\n                  \"accountIndex\": 1,\r\n                  \"accountName\": \"Giro Ind Corp\",\r\n                  \"accountNumber\": \"4599{{ faker 'finance.account' 12 }}\",\r\n                  \"accountStatus\": \"Not Activated\",\r\n                  \"availableBalance\": 403772,\r\n                  \"accountType\": \"CDA\",\r\n                  \"description\": \"\",\r\n                  \"productId\": \"467908\",\r\n                  \"accountBalance\": 4037.72,\r\n                  \"currency\": \"USD\",\r\n                  \"customerName\": \"CN003475 EMI\",\r\n                  \"allowed\": false,\r\n                  \"sharia\": false,\r\n                  \"applicationType\": null\r\n               },\r\n               \"2\":          {\r\n                  \"accountIndex\": 2,\r\n                  \"accountName\": \"SHARIA INIH\",\r\n                  \"accountNumber\": \"4599{{ faker 'finance.account' 12 }}\",\r\n                  \"accountStatus\": \"Blocked\",\r\n                  \"availableBalance\": 1153939,\r\n                  \"accountType\": \"CDA\",\r\n                  \"description\": \"\",\r\n                  \"productId\": \"528919\",\r\n                  \"accountBalance\": 11539.39,\r\n                  \"currency\": \"USD\",\r\n                  \"customerName\": \"CN003475 EMI\",\r\n                  \"allowed\": false,\r\n                  \"sharia\": true,\r\n                  \"applicationType\": \"B\"\r\n               },\r\n               \"3\":          {\r\n                  \"accountIndex\": 3,\r\n                  \"accountName\": \"Dummy BizCard 1\",\r\n                  \"accountNumber\": \"4599{{ faker 'finance.account' 12 }}\",\r\n                  \"accountStatus\": \"Active\",\r\n                  \"availableBalance\": 101450093333,\r\n                  \"accountType\": \"CDA\",\r\n                  \"description\": \"\",\r\n                  \"productId\": \"428416\",\r\n                  \"accountBalance\": 1.01450093333E9,\r\n                  \"currency\": \"IDR\",\r\n                  \"customerName\": \"CN003475 EMI\",\r\n                  \"allowed\": false,\r\n                  \"sharia\": false,\r\n                  \"applicationType\": \"B\"\r\n               },\r\n               \"4\":          {\r\n                  \"accountIndex\": 4,\r\n                  \"accountName\": \"Giro Ind Corp\",\r\n                  \"accountNumber\": \"4599{{ faker 'finance.account' 12 }}\",\r\n                  \"accountStatus\": \"Active\",\r\n                  \"availableBalance\": 847612355,\r\n                  \"accountType\": \"CDA\",\r\n                  \"description\": \"\",\r\n                  \"productId\": \"441659\",\r\n                  \"accountBalance\": 8476123.55,\r\n                  \"currency\": \"IDR\",\r\n                  \"customerName\": \"CN003475 EMI\",\r\n                  \"allowed\": false,\r\n                  \"sharia\": false,\r\n                  \"applicationType\": null\r\n               },\r\n               \"5\":          {\r\n                  \"accountIndex\": 5,\r\n                  \"accountName\": \"Giro Ind Corp\",\r\n                  \"accountNumber\": \"4599{{ faker 'finance.account' 12 }}\",\r\n                  \"accountStatus\": \"Active\",\r\n                  \"availableBalance\": 847612355,\r\n                  \"accountType\": \"CDA\",\r\n                  \"description\": \"\",\r\n                  \"productId\": \"548117\",\r\n                  \"accountBalance\": 8476123.55,\r\n                  \"currency\": \"IDR\",\r\n                  \"customerName\": \"CN003475 EMI\",\r\n                  \"allowed\": false,\r\n                  \"sharia\": false,\r\n                  \"applicationType\": \"B\"\r\n               },\r\n               \"6\":          {\r\n                  \"accountIndex\": 6,\r\n                  \"accountName\": \"Giro Ind Corp\",\r\n                  \"accountNumber\": \"4599{{ faker 'finance.account' 12 }}\",\r\n                  \"accountStatus\": \"Active\",\r\n                  \"availableBalance\": 847612355,\r\n                  \"accountType\": \"CDA\",\r\n                  \"description\": \"\",\r\n                  \"productId\": \"459920\",\r\n                  \"accountBalance\": 8476123.55,\r\n                  \"currency\": \"IDR\",\r\n                  \"customerName\": \"CN003475 EMI\",\r\n                  \"allowed\": false,\r\n                  \"sharia\": false,\r\n                  \"applicationType\": \"B\"\r\n               },\r\n               \"7\":          {\r\n                  \"accountIndex\": 7,\r\n                  \"accountName\": \"Giro Ind Corp\",\r\n                  \"accountNumber\": \"4599{{ faker 'finance.account' 12 }}\",\r\n                  \"accountStatus\": \"Active\",\r\n                  \"availableBalance\": 847612355,\r\n                  \"accountType\": \"CDA\",\r\n                  \"description\": \"\",\r\n                  \"productId\": \"533619\",\r\n                  \"accountBalance\": 8476123.55,\r\n                  \"currency\": \"IDR\",\r\n                  \"customerName\": \"CN003475 EMI\",\r\n                  \"allowed\": false,\r\n                  \"sharia\": false,\r\n                  \"applicationType\": null\r\n               },\r\n               \"8\": {\r\n                  \"accountIndex\": 8,\r\n                  \"accountName\": \"Wave n Go Sticker Visa\",\r\n                  \"accountNumber\": \"4599{{ faker 'finance.account' 12 }}\",\r\n                  \"accountStatus\": \"Active\",\r\n                  \"availableBalance\": 847612355,\r\n                  \"accountType\": \"CDA\",\r\n                  \"description\": \"\",\r\n                  \"productId\": \"457941\",\r\n                  \"accountBalance\": 8476123.55,\r\n                  \"currency\": \"IDR\",\r\n                  \"customerName\": \"CN003475 EMI\",\r\n                  \"allowed\": false,\r\n                  \"sharia\": false,\r\n                  \"applicationType\": \"B\"\r\n               },\r\n               \"9\": {\r\n                  \"accountIndex\": 9,\r\n                  \"accountName\": \"Wave n Go Plastic Visa\",\r\n                  \"accountNumber\": \"4599{{ faker 'finance.account' 12 }}\",\r\n                  \"accountStatus\": \"Active\",\r\n                  \"availableBalance\": 847612355,\r\n                  \"accountType\": \"CDA\",\r\n                  \"description\": \"\",\r\n                  \"productId\": \"457942\",\r\n                  \"accountBalance\": 8476123.55,\r\n                  \"currency\": \"IDR\",\r\n                  \"customerName\": \"CN003475 EMI\",\r\n                  \"allowed\": false,\r\n                  \"sharia\": false,\r\n                  \"applicationType\": \"B\"\r\n               }\r\n            },\r\n            \"caaccountList\": {},\r\n            \"saaccountList\": {},\r\n            \"nonaccountList\": {},\r\n            \"loanaccountList\": {}\r\n        }\r\n    }\r\n}",
                      "latency": 0,
                      "statusCode": 200,
                      "headers": [
                          {
                              "key": "",
                              "value": ""
                          }
                      ],
                      "filePath": "",
                      "sendFileAsBody": false,
                      "rules": [
                          {
                              "target": "body",
                              "modifier": "pagination",
                              "value": "Y",
                              "isRegex": false
                          }
                      ],
                      "label": "All types included",
                      "disableTemplating": false,
                      "rulesOperator": "OR"
                  },
                  {
                      "uuid": "",
                      "body": "{\r\n  \"apierror\": {\r\n    \"status\": \"INTERNAL_SERVER_ERROR\",\r\n    \"timestamp\": \"09-09-2019 01:01:01\",\r\n    \"errorTitle\": \"Internal Server Error\",\r\n    \"message\": \"System Malfunction - Please contact CIMB Niaga Customer Service 14041\",\r\n    \"errorCode\": \"T1\",\r\n    \"debugMessage\": null,\r\n    \"sourceSystem\": \"SICS_ID\",\r\n    \"errorTitleIdn\": \"Kesalahan server dari dalam\",\r\n    \"errorMessageIdn\": \"Kerusakan Sistem - Silakan hubungi Layanan Pelanggan CIMB Niaga 14041\",\r\n    \"errorTitleEng\": \"Internal Server Error\",\r\n    \"errorMessageEng\": \"System Malfunction - Please contact CIMB Niaga Customer Service 14041\",\r\n    \"traceId\": \"78262ehjl919e126\"\r\n  }\r\n}",
                      "latency": 0,
                      "statusCode": 400,
                      "headers": [
                          {
                              "key": "",
                              "value": ""
                          }
                      ],
                      "filePath": "",
                      "sendFileAsBody": false,
                      "rules": [],
                      "label": "No Account",
                      "disableTemplating": false,
                      "rulesOperator": "OR"
                  },
                  {
                      "uuid": "",
                      "body": "{\n    \"error\": \"invalid_token\",\n    \"error_description\": \"9F0D9D63D5ACD42A551FCB86F9E5C10CD5460D7B2CDF7126ED11D2A7947F6A41\"\n}",
                      "latency": 0,
                      "statusCode": 401,
                      "label": "",
                      "headers": [
                          {
                              "key": "",
                              "value": ""
                          }
                      ],
                      "filePath": "",
                      "sendFileAsBody": false,
                      "rules": [],
                      "disableTemplating": false,
                      "rulesOperator": "OR"
                  },
                  {
                      "uuid": "",
                      "body": "{\r\n  \"apierror\": {\r\n    \"status\": \"INTERNAL_SERVER_ERROR\",\r\n    \"timestamp\": \"09-09-2019 01:01:01\",\r\n    \"errorTitle\": \"Internal Server Error\",\r\n    \"message\": \"System Malfunction - Please contact CIMB Niaga Customer Service 14041\",\r\n    \"errorCode\": \"8888\",\r\n    \"debugMessage\": null,\r\n    \"sourceSystem\": \"SIBS\",\r\n    \"errorTitleIdn\": \"Kesalahan server dari dalam\",\r\n    \"errorMessageIdn\": \"Kerusakan Sistem - Silakan hubungi Layanan Pelanggan CIMB Niaga 14041\",\r\n    \"errorTitleEng\": \"Internal Server Error\",\r\n    \"errorMessageEng\": \"System Malfunction - Please contact CIMB Niaga Customer Service 14041\",\r\n    \"traceId\": \"78262ehjl919e126\"\r\n  }\r\n}",
                      "latency": 0,
                      "statusCode": 500,
                      "headers": [
                          {
                              "key": "",
                              "value": ""
                          }
                      ],
                      "filePath": "",
                      "sendFileAsBody": false,
                      "rules": [],
                      "label": "",
                      "disableTemplating": false,
                      "rulesOperator": "OR"
                  }
              ],
              "enabled": true,
              "randomResponse": false
          }
      }
  ]
}`

**Comment by 255kb** [comment, marked as answer]
Hi @alvinchristian7 
Mockoon can return a 500 in various cases, but is usually quit tolerant and this does not happen often. 
Could you please export your settings (Import/Export->Mockoon format->Export to JSON) and add it here together with an example call (method, path, headers, body) so I can investigate?

**Reply by alvinchristian7** [reply]
ok edited

**Reply by 255kb** [reply]
Apparently, response rules interpreter is failing when sending a null value in the body. 
I opened an issue: https://github.com/mockoon/commons/issues/16
It will be fixed for the next release (due in February)

**Reply by alvinchristian7** [reply]
ok thanks
